vllm/config.py

```diff
@@ -141,6 +141,18 @@ def __init__(
                                     code_revision, rope_scaling, rope_theta)
         self.hf_text_config = get_hf_text_config(self.hf_config)
         self.dtype = _get_and_verify_dtype(self.hf_text_config, dtype)
+
+        if (not self.disable_sliding_window
+                and self.hf_text_config.model_type == "gemma2"
+                and self.hf_text_config.sliding_window is not None):
+            print_warning_once(
+                "While Gemma 2 uses sliding window attention for every odd "
+                "layer, vLLM currently ignores it and uses global attention "
+                "for all layers. This might affect the model's behavior when "
+                "the context length is larger than the sliding window size "
+                f"({self.hf_text_config.sliding_window}).")
```

Comment:
Changes look good.

I think this warning should be updated to say something like `Gemma 2 uses sliding window attention for every odd layer, which is not supported by vllm. Disabling sliding window and capping max length to sliding_window_size`

Reply:
@robertgshaw2-neuralmagic Oh maybe I misunderstood the change here. My intention was to enable the full (8K) context length with global attention for all layers.

Reply:
Okay sorry for the confusion the way you had it before will do this :)

Setting `disable_sliding_window=True` will cap to `sliding_window_size`

Reply:
I think either option is reasonable.

- For capping at 4k, its more conservative re: model accuracy
- For capping at 8k, its less conservative re: model accuracy

Reply:
@robertgshaw2-neuralmagic Hmm... OK let's use 4K context length for now and see if people want 8K content length despite the difference from the original model.

Reply:
@robertgshaw2-neuralmagic Updated the warning msg. PTAL!

Reply:
LGTM